src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/Inflater.cs

```diff
@@ -313,14 +322,16 @@ private ZLibNative.ErrorCode Inflate(ZLibNative.FlushCode flushCode)
         /// </summary>
         private void DeallocateInputBufferHandle()
         {
-            Debug.Assert(_inputBufferHandle.IsAllocated);
+            Debug.Assert(IsInputBufferHandleAllocated);
 
             lock (SyncLock)
             {
                 _zlibStream.AvailIn = 0;
                 _zlibStream.NextIn = ZLibNative.ZNullPtr;
-                _inputBufferHandle.Free();
+                _inputBufferHandle.Dispose();
             }
         }
+
+        private unsafe bool IsInputBufferHandleAllocated => _inputBufferHandle.Pointer != (void*)0;
```

Comment:
we should be able to use `default` here (and make the code look less like `C` lang ;) )

```suggestion
        private unsafe bool IsInputBufferHandleAllocated => _inputBufferHandle.Pointer != default;
```